add reorder event and callback

kyle-mccarthy opened the pull request.

rikschennink commented:
Thanks @kyle-mccarthy, I'm reviewing.

rikschennink commented on src/js/app/actions.js:
```diff
@@ -207,6 +207,7 @@ export const actions = (dispatch, query, state) => ({
         index = limit(index, 0, state.items.length - 1);
         if (currentIndex === index) return;
         state.items.splice(index, 0, state.items.splice(currentIndex, 1)[0]);
+        dispatch('DID_REORDER_ITEMS', { items: getActiveItems(state.items) });
```
This will also fire if the order is changed using the FilePond API, not sure if that is an issue perse, if it is, we need to move this to the drag item method.

rikschennink commented on types/index.d.ts:
```diff
@@ -701,7 +703,8 @@ type FilePondEventPrefixed = 'FilePond:init'
     | 'FilePond:removefile'
-    | 'FilePond:updatefiles';
+    | 'FilePond:updatefiles'
+    | 'FilePond:reorderedfiles';
```
I think this needs to be `reorderfiles` instead of `reorderedfiles `

rikschennink commented on types/index.d.ts:
```diff
@@ -715,7 +718,8 @@ type FilePondEvent = 'init'
     | 'removefile'
-    | 'updatefiles';
+    | 'updatefiles'
+    | 'reorderedfiles';
```
I think this needs to be `reorderfiles` instead of `reorderedfiles`

kyle-mccarthy commented:
You are correct about the type defs and I fixed those. I also moved the dispatch event into the view like you suggested. 

One thing is that it will trigger the event while the user is still moving the items. What I mean by that is if there are two files A and B, if you click and move them to B and A and then move them back to A and B without releasing the mouse it will fire the event twice. 

I am not sure if this is a problem or not, but I don't think that there is a easy way to resolve it. As far as I can tell the state of the mouse isn't currently tracked so that would have to be added.

rikschennink commented:
I think that's fine for now, at that point the file's are actually re-ordered in the model. I'll merge this later this week.

ersanserkan commented:
Thanks @kyle-mccarthy and @rikschennink ,

By the way, it is missing in documentations.
